### PR TITLE
remove versioned printer

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Given that we now have a way to fail if internal objects are passed to printers, we can ensure that object conversion is done by consumers of printers. Therefore, the versioned printer is no longer needed.

Depends on https://github.com/kubernetes/kubernetes/pull/63822

cc @deads2k @soltysh

### CODE REVIEW

**Comment by deads2k** [comment]
wait for e2es to finish. If that's it, let's push through.

**Comment by deads2k** [comment]
looks like you missed a test.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve no-issue

File: pkg/kubectl/cmd/create/create.go [hunk 1]
```diff
@@ -445,5 +445,5 @@ func (o *CreateSubcommandOptions) Run() error {
 		}
 	}
 
-	return o.PrintObj(obj)
+	return o.PrintObj(cmdutil.AsDefaultVersionedOrOriginal(obj, nil))
```
[the remark this answers is not in the document]
Comment (juanvallejo) [reply pkg/kubectl/cmd/create/create.go]: Not sure if we should be adding the TypeMeta by hand in the generator. Here, for example: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/secret.go#L132

File: pkg/kubectl/cmd/create/create_clusterrole.go [hunk 2]
```diff
@@ -164,13 +164,15 @@ func (c *CreateClusterRoleOptions) RunCreateRole() error {
 	}
 	clusterRole.Rules = rules
 
+	if c.DryRun {
```
Comment (soltysh) [inline pkg/kubectl/cmd/create/create_clusterrole.go]: I think that's reasonable approach.

**Comment by deads2k** [comment]
/hold cancel

/lgtm

We'll use this pull to get most of the dep snipped.  I'll come at kubectl name printer another way.

@juanvallejo you need to get green.

**Comment by juanvallejo** [comment]
@deads2k fixed integration test failure

File: pkg/kubectl/cmd/get/get.go [hunk 3]
```diff
@@ -420,6 +418,13 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 			lastMapping = mapping
 		}
 
+		// ensure a versioned object is passed to the custom-columns printer
```
Comment (deads2k) [inline pkg/kubectl/cmd/get/get.go]: @soltysh ptal at this

Comment (soltysh) [reply pkg/kubectl/cmd/get/get.go]: That's not needed, this should work nicely with the next ifs, @juanvallejo any particular reason for doing this?

**Review by soltysh — APPROVED** [review]
The changes itself lgtm, except for that one question of mine.

**Comment by soltysh** [comment]
Closed in favor of #63928.

**Pull request closed without merging**